examples/tokenvm/avalanche-ops.aws.md

```diff
@@ -0,0 +1,216 @@
+
+```bash
+# run the following to create resources
+vi /home/ubuntu/aops-custom-202303-vzLY7T.yaml
+
+avalancheup-aws apply \
+--spec-file-path /home/ubuntu/aops-custom-202303-vzLY7T.yaml
+
+# run the following to delete resources
+avalancheup-aws delete \
+--delete-cloudwatch-log-group \
+--delete-s3-objects \
+--delete-ebs-volumes \
+--delete-elastic-ips \
+--spec-file-path /home/ubuntu/aops-custom-202303-vzLY7T.yaml
+```
+
+`apply` creates AWS resources, whereas `delete` destroys after testing is done.
+
+NOTE: You may encounter [aws-sdk-rust#611](https://github.com/awslabs/aws-sdk-rust/issues/611) where AWS SDK call hangs, which blocks node bootstraps. If a node takes too long to start, connect to that instance (use SSM sesson from AWS console), and restart the agent with the command `sudo systemctl restart avalanched-aws`.
+
+### Step 4
+
+Now that the network and nodes are up, we can install two subnets, each of which runs its own TokenVM. A successful `avalancheup-aws apply` run will output the commands as follow (which you can customize depending on your VM):
+
+```bash
+# EXAMPLE: write subnet config
+avalancheup-aws subnet-config \
+--log-level=info \
+--proposer-min-block-delay 250000000 \
+--file-path /tmp/subnet-config.json
+
+# EXAMPLE: install subnet-evm in all nodes
+avalancheup-aws install-subnet-chain \
+--log-level info \
+--region us-west-2 \
+--s3-bucket avalanche-ops-202303-3hxfve9pfh-us-west-2 \
+--s3-key-prefix aops-custom-202303-2qUWt5/install-subnet-chain \
+--ssm-doc aops-custom-202303-2qUWt5-ssm-install-subnet-chain \
+--chain-rpc-url http://52.88.8.107:9650 \
+...
+```
+
+For instance, use the following commands to deploy two TokenVM subnets (similar to [scripts/run.sh](./scripts/run.sh)):
+
+```bash
+avalancheup-aws subnet-config \
+--log-level=info \
+--proposer-min-block-delay 0 \
+--file-path /tmp/subnet-config.json
+
+cat <<EOF > /tmp/allocations.json
+[{"address":"token1rvzhmceq997zntgvravfagsks6w0ryud3rylh4cdvayry0dl97nsjzf3yp", "balance":1000000000000}]
+EOF
+rm -f /tmp/tokenvm-genesis.json
+/tmp/token-cli genesis generate /tmp/allocations.json \
+--genesis-file /tmp/tokenvm-genesis.json
+cat /tmp/tokenvm-genesis.json
+
+cat <<EOF > /tmp/tokenvm-chain-config.json
+{
+  "mempoolSize": 10000000,
+  "mempoolExemptPayers":["token1rvzhmceq997zntgvravfagsks6w0ryud3rylh4cdvayry0dl97nsjzf3yp"],
+  "parallelism": 5,
+  "trackedPairs":["*"],
+  "logLevel": "info",
+  "stateSyncServerDelay": 0
+}
+EOF
+cat /tmp/tokenvm-chain-config.json
+```
+
+```bash
+avalancheup-aws install-subnet-chain \
+--log-level info \
+--region us-west-2 \
+--s3-bucket avalanche-ops-202303-3hxfve9pfh-us-west-2 \
+--s3-key-prefix aops-custom-202303-2qUWt5/install-subnet-chain \
+--ssm-doc aops-custom-202303-2qUWt5-ssm-install-subnet-chain \
+--chain-rpc-url http://52.88.8.107:9650 \
+--key 0x56289e99c94b6912bfc12adc093c9b51124f0dc54ac7a766b2bc5ccf558d8027 \
+--staking-perioid-in-days 15 \
```

Comment:
Does this also register BLS public keys when issuing initial staking transactions?

Reply:
Got this working without changing any end-user side command (just switched to `AddPermissionlessValidatorTx`, see https://github.com/ava-labs/avalanche-ops/pull/277). Will update the doc.